workshops/evals/evals.ipynb

```diff
@@ -0,0 +1,757 @@
+                "\"\"\"\n",
+                "# Hide: none\n",
+                "print(llm_as_judge_prompt)"
+            ]
+        },
+        {
+            "cell_type": "markdown",
+            "metadata": {},
+            "source": [
+                "### Running the evals\n",
+                "\n",
+                "The three pieces above - the dataset, the plan of solvers and the scorer - are what define a single evaluation task.\n",
+                "This task can then be run on the models you wish to evaluate."
+            ]
+        },
+        {
+            "cell_type": "code",
+            "execution_count": null,
+            "metadata": {},
+            "outputs": [],
+            "source": [
+                "# to define a task, we apply the @task decorator on a function that returns a Task object.\n",
+                "# Note we use `dataset[0:3]` to only use the first 3 samples in the dataset.\n",
+                "# This speeds up the notebook and is good practice when developing new tasks or for testing the workflow.\n",
+                "@task\n",
+                "def theory_of_mind(model: str):\n",
+                "    return Task(\n",
+                "        dataset=dataset[0:3],\n",
+                "        plan=[\n",
+                "            chain_of_thought(),\n",
+                "            generate(),\n",
+                "            self_critique(model=model),\n",
+                "        ],\n",
+                "        scorer=scorer,\n",
+                "    )"
+            ]
+        },
+        {
+            "cell_type": "code",
+            "execution_count": null,
+            "metadata": {},
+            "outputs": [],
+            "source": [
+                "# To evaluate a model on the task, we use the `eval` function\n",
+                "model = \"openai/gpt-4o-mini\"\n",
+                "logs = eval(\n",
+                "    theory_of_mind(model),\n",
+                "    model=model,\n",
+                ")"
+            ]
+        },
+        {
+            "cell_type": "markdown",
+            "metadata": {},
+            "source": [
+                "Using the `eval` function is one of the adjustments made to adapt inspect to a notebook.\n",
+                "The standard way to evaluate a model with Inspect is to define the task in a python file and then run the `eval` command in a CLI.\n",
+                "See their [docs](https://inspect.ai-safety-institute.org.uk/#getting-started) for an example.\n",
+                "\n",
+                "The second adjustment is that Inspect includes a function `view` that creates a simple browser-based GUI to view the results of an evaluation.\n",
+                "As far as we know, this does not work with Google Colab so we have to manually define code to parse the `logs` outputted by the evals."
```

Comment:
I looked at it quickly, and it seems doable in 1h of my time, but I'm not sure it's worth the effort. What do you think?

Reply:
Ideally this would be nice. However, setting up servers is something I know nothing about and presumably many others too. Hence, this could lead to a maintenance problem where you are not available to fix things. My preference is to keep things as is.